Comparison operators were not declared const, so compilation error 'error: static assertion failed: comparison object must be invocable as const' with toolchain gfbf-2024a
Author: Douglas G Scofield (UPPMAX)
diff -ruN plink-1.07-src_orig/genogroup.h plink-1.07-src/genogroup.h
--- plink-1.07-src_orig/genogroup.h	2009-10-10 18:00:22.000000000 +0200
+++ plink-1.07-src/genogroup.h	2025-06-13 17:18:37.036371000 +0200
@@ -42,7 +42,7 @@
   template<> class less<MultiLocusGenotype*> {
   public:
     bool operator()(MultiLocusGenotype const* p1, 
-		    MultiLocusGenotype const* p2) {
+                    MultiLocusGenotype const* p2) const {
       if (!p1)
 	return true;
       if (!p2)
diff -ruN plink-1.07-src_orig/homozyg.cpp plink-1.07-src/homozyg.cpp
--- plink-1.07-src_orig/homozyg.cpp	2009-10-10 18:00:21.000000000 +0200
+++ plink-1.07-src/homozyg.cpp	2025-06-13 17:00:31.653445000 +0200
@@ -33,7 +33,7 @@
   template<>
   class less<Segment*> {
   public:
-  bool operator()(Segment const* s1, Segment const* s2)
+  bool operator()(Segment const* s1, Segment const* s2) const
   {
     if      ( s1->start  > s2->start ) return true;
     else if ( s1->start  < s2->start ) return false;
@@ -70,7 +70,7 @@
   template<>
   class less<Pool*> {
   public:
-  bool operator()(Pool const* p1, Pool const* p2)
+  bool operator()(Pool const* p1, Pool const* p2) const
   {
 
     if      ( p1->segs.size() > p2->segs.size() ) return true; 
diff -ruN plink-1.07-src_orig/plink.h plink-1.07-src/plink.h
--- plink-1.07-src_orig/plink.h	2009-10-10 18:00:22.000000000 +0200
+++ plink-1.07-src/plink.h	2025-06-13 15:57:48.627689000 +0200
@@ -387,7 +387,7 @@
   template<>
     class less<Locus*> {
     public:
-    bool operator()(Locus const* p1, Locus const* p2)
+    bool operator()(Locus const* p1, Locus const* p2) const
       {
 
 	// Locus comparison based first on distance, 
